src/features/auth/pages/login/Login.native.test.tsx

```diff
@@ -320,19 +279,91 @@ describe('<Login/>', () => {
 
     expect(analytics.logSignUp).toHaveBeenNthCalledWith(1, { from: 'Login' })
   })
+
+  describe('Login comes from an offer', () => {
+    const OFFER_ID = favoriteResponseSnap.offer.id
+    beforeEach(() => {
+      useRoute
+        .mockReturnValueOnce({ params: { offerId: OFFER_ID } }) // first render
+        .mockReturnValueOnce({ params: { offerId: OFFER_ID } }) // email input rerender
+        .mockReturnValueOnce({ params: { offerId: OFFER_ID } }) // password input rerender
+    })
+
+    it('should redirect to the previous offer page when signin is successful', async () => {
+      const renderAPI = renderLogin()
+      fillInputs(renderAPI)
+      fireEvent.press(renderAPI.getByText('Se connecter'))
+
+      await waitFor(() => {
+        expect(navigate).toHaveBeenNthCalledWith(1, 'Offer', {
+          id: OFFER_ID,
+        })
+      })
+    })
+
+    it('should add the previous offer to favorites when signin is successful', async () => {
+      simulateAddToFavorites()
+
+      const renderAPI = renderLogin()
+      fillInputs(renderAPI)
+      fireEvent.press(renderAPI.getByText('Se connecter'))
+
+      await waitFor(() => {
+        expect(mockPostFavorite).toHaveBeenCalledTimes(1)
+      })
+    })
+
+    it('should log analytics when adding the previous offer to favorites', async () => {
+      simulateAddToFavorites()
+      const renderAPI = renderLogin()
+      fillInputs(renderAPI)
+      fireEvent.press(renderAPI.getByText('Se connecter'))
+
+      await waitFor(() => {
+        expect(analytics.logHasAddedOfferToFavorites).toHaveBeenCalledWith({
+          from: 'login',
+          offerId: OFFER_ID,
+        })
+      })
+    })
+
+    it('should redirect to CulturalSurveyIntro instead of Offer when user needs to fill it and users comes from offer modal', async () => {
```

Comment:
il manque ce test à renommer

```suggestion
    it('should redirect to CulturalSurveyIntro instead of Offer when user needs to fill it', async () => {
```